alerting/src/main/kotlin/org/opensearch/alerting/MonitorMetadataService.kt

```diff
@@ -208,10 +216,18 @@ object MonitorMetadataService :
         createWithRunContext: Boolean,
         workflowMetadataId: String? = null,
     ): MonitorMetadata {
-        val monitorIndex = if (monitor.monitorType == Monitor.MonitorType.DOC_LEVEL_MONITOR)
+        val monitorIndex = if (
+            monitor.isMonitorOfStandardType() &&
+            Monitor.MonitorType.valueOf(monitor.monitorType.uppercase(Locale.ROOT)) == Monitor.MonitorType.DOC_LEVEL_MONITOR
+        )
             (monitor.inputs[0] as DocLevelMonitorInput).indices[0]
         else null
-        val runContext = if (monitor.monitorType == Monitor.MonitorType.DOC_LEVEL_MONITOR && createWithRunContext)
+        val runContext = if (
+            monitor.isMonitorOfStandardType() &&
```

Comment:
extract into method

Reply:
simplified this logic https://github.com/opensearch-project/alerting/pull/1547/files#diff-5d88a62bc4191f374a4813c2256646402800ae1beed484e7a1812b7445957308R191